business/checkers/destinationrules/no_dest_checker.go

```diff
@@ -77,21 +77,29 @@ func (n NoDestinationChecker) hasMatchingWorkload(host kubernetes.Host, subsetLa
 		}
 	}
 
-	// Check workloads
-	if len(selectors) == 0 {
-		return false
-	}
-
-	selector := labels.SelectorFromSet(labels.Set(selectors))
-
 	subsetLabelSet := labels.Set(subsetLabels)
 	subsetSelector := labels.SelectorFromSet(subsetLabelSet)
 
-	for _, wl := range n.WorkloadsPerNamespace[localNs].Workloads {
-		wlLabelSet := labels.Set(wl.Labels)
-		if selector.Matches(wlLabelSet) {
-			if subsetSelector.Matches(wlLabelSet) {
-				return true
+	// Check workloads
```

Comment:
The code looks good.
About naming, perhaps could be good to rename the upper meeting "hasMatchingWorkload" to "hasMatchingEndpoint" in the meaning of Istio:
![image](https://user-images.githubusercontent.com/1662329/172355359-98ce3592-08b3-46e0-9a8f-ec7d58d4cf79.png)

It's just to not forget that the DR could point to any service and not necessarily a workload (we don't have that supported in other places, like the wizards but it can be a start).